application/di/init.go

```diff
@@ -119,7 +122,8 @@ func initInfrastructure(c *config.Config) {
 	instrumentor = performance2.NewInstrumentor()
 	snykApiClient = snyk_api.NewSnykApiClient(c, networkAccess.GetHttpClient)
 	gafConfiguration := c.Engine().GetConfiguration()
-
+	scanPersister = persistence.NewGitPersistenceProvider(c.Logger(), afero.NewOsFs())
+	scanPersister.Init()
```

Comment:
let's not call it here, but during initialization - we need to register it as initializer, see cliInitializer and authInitializer